### PR TITLE
Added CBMC proof of OTA_HTTP_strerror function

### PR DESCRIPTION
**Pull request by Bujain**
*Description of changes:*
Added CBMC-proof files of OTA_HTTP_strerror function. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/OTA_HTTP_strerror/cbmc-viewer.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{ "expected-missing-functions":
+  [
+
```
Comment (yanjos-dev) [inline test/cbmc/proofs/OTA_HTTP_strerror/cbmc-viewer.json:3]: Please remove the extra whitespace.

File: test/cbmc/proofs/OTA_HTTP_strerror/OTA_HTTP_strerror_harness.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+void OTA_HTTP_strerror_harness()
+{
+    OtaHttpStatus_t status;
+    const char * pErrorMsg = NULL;
+
+    pErrorMsg = OTA_HTTP_strerror( status );
+
+    /* The function returns a string which is never a NULL. If it does, then
+     *  there is a problem. */
+    __CPROVER_assert( pErrorMsg != NULL, "function cannot return null" );
```
Comment (yanjos-dev) [inline test/cbmc/proofs/OTA_HTTP_strerror/OTA_HTTP_strerror_harness.c:40]: Is "function cannot return null" the error log that will be printed if this is triggered?

If so, please update this to be a complete sentence and to be more clear about what the failure was compared to what was expected.

**Review by yanjos-dev — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/OTA_HTTP_strerror/cbmc-viewer.json (hunk 1 above)
Comment (Bujain) [reply test/cbmc/proofs/OTA_HTTP_strerror/cbmc-viewer.json]: Removed it in 
6f40538

File: test/cbmc/proofs/OTA_HTTP_strerror/OTA_HTTP_strerror_harness.c (hunk 2 above)
Comment (Bujain) [reply test/cbmc/proofs/OTA_HTTP_strerror/OTA_HTTP_strerror_harness.c]: Updated it in 
6f40538

**Review by yanjos-dev — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**